plugins/commands/cloud/version/release.rb

```diff
@@ -15,58 +17,56 @@ def execute
             return if !argv
-            if argv.empty? || argv.length > 2
+            if argv.size != 2
               raise Vagrant::Errors::CLIInvalidUsage,
                 help: opts.help.chomp
             end
 
-            if not options[:force]
-                @env.ui.warn(I18n.t("cloud_command.version.release_warn", version: argv[1], box: argv.first))
-                cont = @env.ui.ask(I18n.t("cloud_command.continue"))
-                return 1 if cont.strip.downcase != "y"
+            if !options[:force]
+              @env.ui.warn(I18n.t("cloud_command.version.release_warn", version: argv[1], box: argv.first))
+              cont = @env.ui.ask(I18n.t("cloud_command.continue"))
+              return 1 if cont.strip.downcase != "y"
             end
 
-            @client = VagrantPlugins::CloudCommand::Util.client_login(@env, options[:username])
-            box = argv.first.split('/', 2)
-            org = box[0]
-            box_name = box[1]
+            @client = client_login(@env)
+            org, box_name = argv.first.split('/', 2)
             version = argv[1]
 
             release_version(org, box_name, version, @client.token, options)
           end
 
-          def release_version(org, box_name, version, access_token, options)
-            org = options[:username] if options[:username]
-
-            server_url = VagrantPlugins::CloudCommand::Util.api_server_url
-            account = VagrantPlugins::CloudCommand::Util.account(org, access_token, server_url)
-            box = VagrantCloud::Box.new(account, box_name, nil, nil, nil, access_token)
-            version = VagrantCloud::Version.new(box, version, nil, nil, access_token)
-
-            begin
-              success = version.release
-              @env.ui.success(I18n.t("cloud_command.version.release_success", version: version, org: org, box_name: box_name))
-              success = success.delete_if{|_, v|v.nil?}
-              VagrantPlugins::CloudCommand::Util.format_box_results(success, @env)
-              return 0
-            rescue VagrantCloud::ClientError => e
-              @env.ui.error(I18n.t("cloud_command.errors.version.release_fail", version: version, org: org, box_name: box_name))
-              @env.ui.error(e)
-              return 1
+          # Release the box version
+          #
+          # @param [String] org Organization name
+          # @param [String] box_name Box name
+          # @param [String] version Version of the box
+          # @param [String] access_token User Vagrant Cloud access token
+          # @param [Hash] options Currently unused
+          # @return [Integer]
+          def release_version(org, box_name, version, access_token, options={})
+            account = VagrantCloud::Account.new(
+              custom_server: api_server_url,
+              access_token: access_token
+            )
+            with_version(account: account, org: org, box: box_name, version: version) do |v|
+              v.release
```

Comment:
should probably have some error checking here. Trying to release a box with no providers uploaded results in an error

```
bundler: failed to load command: vagrant (/Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/bin/vagrant)
Excon::Error::UnprocessableEntity: Expected([200, 201, 204]) <=> Actual(422 Unprocessable Entity)

  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/excon-0.78.0/lib/excon/middlewares/expects.rb:13:in `response_call'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/excon-0.78.0/lib/excon/middlewares/response_parser.rb:12:in `response_call'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/excon-0.78.0/lib/excon/connection.rb:456:in `response'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/excon-0.78.0/lib/excon/connection.rb:287:in `request'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/client.rb:129:in `block in request'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/client.rb:84:in `block in with_connection'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/client.rb:84:in `synchronize'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/client.rb:84:in `with_connection'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/client.rb:129:in `request'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/client.rb:343:in `box_version_release'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/gems/vagrant_cloud-3.0.1/lib/vagrant_cloud/box/version.rb:58:in `release'
  /Users/sophia/project/vagrant/plugins/commands/cloud/version/release.rb:60:in `block in release_version'
  /Users/sophia/project/vagrant/plugins/commands/cloud/util.rb:171:in `block in with_version'
  /Users/sophia/project/vagrant/plugins/commands/cloud/util.rb:151:in `with_box'
  /Users/sophia/project/vagrant/plugins/commands/cloud/util.rb:164:in `with_version'
  /Users/sophia/project/vagrant/plugins/commands/cloud/version/release.rb:59:in `release_version'
  /Users/sophia/project/vagrant/plugins/commands/cloud/version/release.rb:43:in `execute'
  /Users/sophia/project/vagrant/plugins/commands/cloud/version/root.rb:50:in `execute'
  /Users/sophia/project/vagrant/plugins/commands/cloud/root.rb:73:in `execute'
  /Users/sophia/project/vagrant/lib/vagrant/cli.rb:67:in `execute'
  /Users/sophia/project/vagrant/lib/vagrant/environment.rb:290:in `cli'
  /Users/sophia/project/vagrant/bin/vagrant:205:in `<top (required)>'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/bin/vagrant:23:in `load'
  /Users/sophia/miniconda3/envs/vagrant/lib/ruby/gems/2.6.0/bin/vagrant:23:in `<top (required)>'
```

Reply:
Fixed via hashicorp/vagrant_cloud#67